b9lab-content/3-main-concepts/14-context.md

```diff
@@ -1 +1,49 @@
+* **Gas Meters**: Specifically, a gasMeter for the transaction currently being processed using the context and a blockGasMeter for the entire block it belongs to. Users specify how much in fees they wish to pay for the execution of their transaction; these gas meters keep track of how much gas has been used in the transaction or block so far. If the gas meter runs out, execution halts.
+* **CheckTx Mode**: A boolean value indicating whether a transaction should be processed in CheckTx or DeliverTx mode.
+* **Min Gas Price**: The minimum gas price a node is willing to take in order to include a transaction in its block. This price is a local value configured by each node individually, and should therefore not be used in any functions used in sequences leading to state-transitions.
+* **Consensus Params**: The ABCI type Consensus Parameters, which specify certain limits for the blockchain, such as maximum gas for a block.
+* **Event Manager**: The event manager allows any caller with access to a Context to emit Events. Modules may define module specific Events by defining various Types and Attributes or use the common definitions found in types/. Clients can subscribe or query for these Events. These Events are collected throughout DeliverTx, BeginBlock, and EndBlock and are returned to Tendermint for indexing. 
+
+## Go Context Package
+
+A basic Context is defined in the Golang Context Package. A Context is an immutable data structure that carries request-scoped data across APIs and processes. Contexts are also designed to enable concurrency and to be used in goroutines. Contexts are intended to be immutable; they should never be edited. Instead, the convention is to create a child context from its parent using a `With` function. The Golang Context Package documentation instructs developers to explicitly pass a context `ctx` as the first argument of a process.
+
+## Store branching
+
+The Context contains a MultiStore, which allows for branchinig and caching functionality using CacheMultiStore (queries in CacheMultiStore are cached to avoid future round trips). 
+
+Each KVStore is branched in a safe and isolated ephemeral storage. Processes are free to write changes to the CacheMultiStore. If a state-transition sequence is performed without issue, the store branch can be committed to the underlying store at the end of the sequence or disregarded if something goes wrong. 
+
+## The pattern of usage for a Context is as follows:
+
+1. process receives a Context `ctx` from its parent process, which provides information needed to perform the process.
+2. The `ctx.ms` is a branched store, meaning a branch of the multistore is made so that the process can make changes to the state as it executes, without changing the original `ctx.ms`. This is useful to protect the underlying multistore in case the changes need to be reverted at some point in the execution.
```

Comment:
```suggestion
2. The `ctx.ms` is a branched store, meaning that a branch of the multistore is made so that the process can make changes to the state as it executes, without changing the original `ctx.ms`. This is useful to protect the underlying multistore in case the changes need to be reverted at some point in the execution.
```